### PR TITLE
Proper RFC3339 support

### PR DESCRIPTION
**Pull request by x0b**
Use a RFC 3339 Parser for modtime parsing as specified by
https://github.com/ncw/rclone/blob/e56be0dfd8bf826fc6b2734032f914da420e4431/cmd/lsjson/lsjson.go#L49
Pull request https://github.com/kaczmarkiewiczp/rcloneExplorer/pull/101 does not correctly parse arbitrary timezone offset strings like ```2018-06-05T00:00:00+01:30```

**Full Disclosure**
The parser library is one of my projects, although with 100% test coverage, so this shouldn't be a problem.

### CODE REVIEW

**Comment by kaczmarkiewiczp** [comment]
Thanks, but for now I'm going to go with #101 . I'm a fan of simple solutions that don't require external libraries when possible.

I also found an issue you might be interested about. Some dates are displayed as "Dec 31 1969". I confirmed this behaviour with three different remotes.

I found the problem to be that your library will throw an `Invalid delimiter` exception, and then the `dateInMillis` is assigned a value of 0. Here's an example of a value that gets passed into your library that causes such an exception
`2018-03-27T17:22:26.14Z`

**Comment by x0b** [comment]
Dammit. Thank you for trying anyways. 

However, considering that non-utc timestamps are still not supported I've reimplemented the timezone parsing and it seems to work so far.
Tested with
- box.com
- dropbox.com
- hubic
- ondrive
- webdav
- crypt remotes

The commit is based on top of #102 to allow conflict-free merging

File: app/src/main/java/ca/pkay/rcloneexplorer/Items/FileItem.java [hunk 1]
```diff
@@ -94,12 +94,22 @@ private String sizeToHumanReadable(long size) {
     }
 
     private long modTimeToMilis(String modTime) {
+        if(modTime.lastIndexOf("+") > 18 || modTime.lastIndexOf("-") > 18){
```
Comment (kaczmarkiewiczp) [inline app/src/main/java/ca/pkay/rcloneexplorer/Items/FileItem.java]: Needs a space between the `if` and opening brace, as well as the closing brace and curly bracket

File: app/src/main/java/ca/pkay/rcloneexplorer/Items/FileItem.java [hunk 2]
```diff
@@ -111,23 +121,27 @@ private long modTimeToMilis(String modTime) {
...
     }
 
-    private String modTimeToHumanReadable(String modTime) {
-        String[] dateTime = modTime.split("T");
-        String formattedDate = dateTime[0] + " " + dateTime[1].substring(0, dateTime[1].length());
-        long now = System.currentTimeMillis();
-        long dateInMillis;
-        Date date;
-        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    private long modTimeZonedToMillis(String modTime){
```
Comment (kaczmarkiewiczp) [inline app/src/main/java/ca/pkay/rcloneexplorer/Items/FileItem.java]: Also space before the opening curly brace

File: app/src/main/java/ca/pkay/rcloneexplorer/Items/FileItem.java [hunk 3]
```diff
@@ -111,23 +121,27 @@ private long modTimeToMilis(String modTime) {
...
+        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX");
+        int index = modTime.lastIndexOf("+");
+        if(index == -1){
```
Comment (kaczmarkiewiczp) [inline app/src/main/java/ca/pkay/rcloneexplorer/Items/FileItem.java]: A space between the `if` and opening brace, and before the opening curly brace.

**Review by patrykcoding — CHANGES_REQUESTED** [review]
Looks like the code is working great now, thanks. There are some styling problems concerning [horizontal whitespace](https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace). I left inline notes. If you could please address that then I'll merge it in. Thanks.

**Review by patrykcoding — APPROVED** [review]

**Pull request merged**